lib/query.js

```diff
@@ -165,6 +165,15 @@ function Query(conditions, options, model, collection) {
   }
 }
 
+// Helper function to check empty/invalid filter
+function checkRequireFilter(filter, options) {
+  if (options && options.requireFilter &&
+      (filter == null ||
+       (typeof filter === 'object' && filter !== null && !Array.isArray(filter) && Object.keys(filter).length === 0))) {
```

Comment:
I don't think `filter` can be an array, so I would get rid of this `Array.isArray(filter)` check.

Also, it would likely be worthwhile to check for empty objects in `$and`, `$or`, and `$nor` as well. For example, `deleteOne({ $and: [{}] })` and `deleteOne({ $or: [{}] })` is equivalent to `deleteOne({})`. That would make `requireFilter` more valuable.